lib/addons/helm-addon/index.ts

```diff
@@ -0,0 +1,19 @@
+import { Construct } from "@aws-cdk/core";
+import * as spi from "../..";
+import { HelmChartConfiguration, KubectlProvider } from "./kubectl-provider";
+
+export type HelmAddOnProps = HelmChartConfiguration;
+export type HelmAddOnUserProps = Partial<HelmChartConfiguration>;
+
+export abstract class HelmAddOn implements spi.ClusterAddOn {
+
+    constructor(readonly props: HelmAddOnProps) {}
+    
+    abstract deploy(clusterInfo: spi.ClusterInfo): void | Promise<Construct>;
+
+    protected addHelmChart(clusterInfo: spi.ClusterInfo, values?: spi.Values ) : Construct {
+        const kubectlProvider = new KubectlProvider(clusterInfo);
+        const chart = {...this.props, ...{ values: values ?? {}} };
```

Comment:
nit - consider moving the ternary out of map literal for ease of reading.

Reply:
Addressed